Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -3464,6 +3464,9 @@ void LocalFrameView::setBaseBackgroundColor(const Color& backgroundColor)
     if (m_baseBackgroundColor == newBaseBackgroundColor)
         return;
 
+#if ENABLE(DARK_MODE_CSS)
+    m_styleColorOptions = styleColorOptions();
+#endif
```

Comment:
I am slightly confused how this change works.

`m_styleColorOptions` is only consulted in `LocalFrameView::recalculateBaseBackgroundColor`, which early returns if the options are equal. Otherwise, it performs this assignment.

So given `m_styleColorOptions` is only read in `recalculateBaseBackgroundColor`, and is always updated there if different, I don't understand how the assignment here has an effect.

Is the real issue that `recalculateBaseBackgroundColor` wipes out the color set by `WebPage::setBackgroundColor`?

Reply:
`m_styleColorOptions` is only read from `recalculateBaseBackgroundColor`, but what I understood from the code is that it assumes that the background won't change if the `m_styleColorOptions` hasn't changed. That is inaccurate since `updateBackgroundRecursively` is called directly from `WebPage::setBackgroundColor`.

> Is the real issue that recalculateBaseBackgroundColor wipes out the color set by WebPage::setBackgroundColor?
I'm not sure, I can see that being a problem, but in the case I was investigating the color is empty

Reply:
Makes sense!